File: Java/14-Ordenação de Vetores/BubbleSort.java
class BubbleSort {
    // Podemos ordenar um vetor através de alguns métodos criados por programadores
    // Este método é chamado BubbleSort
    static int[] bubbleSort(int vetor[]) {
        int tamanho = vetor.length;
        int comparacao = 0;
        int trocas = 0;

        for (int i = tamanho - 1; i >= 1; i--) {
            for (int j = 0; j < i; j++) {

                comparacao++;

                if (vetor[j] > vetor[j + 1]) {

                    int aux = vetor[j];
                    vetor[j] = vetor[j + 1];
                    vetor[j + 1] = aux;
                    trocas++;
                }
            }
        }

        return vetor;
    }

    public static void main(String[] args) {
        // Vetor esta fora de ordem
        int[] vet = { 4, 2, 1, 8, 3 };

        System.out.println("Vetor original");
        for (int v : vet) {
            System.out.print(v + " ");
        }

        System.out.println();

        // Vamos invocar o método passando vetor vet como argumento
        bubbleSort(vet);
        System.out.println("Apos Bubblesort");
        for (int v : vet) {
            System.out.print(v + " ");
        }
    }
}

File: Python/15-Ordenar Vetores/BubbleSort.py
#Podemos ordenar um vetor através de alguns métodos criados por programadores
#Este método é chamado BubbleSort

def bubble_sort(vetor) :
    tamanho = len(vetor) - 1
    ordenado = False
    while not ordenado :
        ordenado = True
        for i in range(tamanho) :
            if vetor[i] > vetor[i + 1] :
                vetor[i], vetor[i + 1] = vetor[i + 1], vetor[i]
                ordenado = False
                print(vetor)
    return vetor

#Criando vetor
vet = [9,5,7,3,1]
print("Vetor original")

for v in vet :
    print(v, end = " ")

print("\n===========")

#Ordenando com bubble sort
print("Bubble Sort")
bubble_sort(vet)
